dart/lib/src/scope.dart

```diff
@@ -213,6 +213,26 @@ class Scope {
       extra.map((key, value) => MapEntry(key, value))
         ..addAll(event.extra ?? {});
 
+  /// If scope and event have a user, the user of the event takes
+  /// precedence.
+  SentryUser? _mergeUsers(SentryUser? scopeUser, SentryUser? eventUser) {
+    if (scopeUser == null && eventUser != null) {
+      return eventUser;
+    }
+    if (eventUser == null && scopeUser != null) {
+      return scopeUser;
+    }
+    // otherwise the user of scope takes precedence over the event user
+    return scopeUser?.copyWith(
+      id: eventUser?.id,
+      email: eventUser?.email,
+      ipAddress: eventUser?.ipAddress,
+      username: eventUser?.username,
+      // TODO should extras be merged?
+      extras: eventUser?.extras,
```

Comment:
I don't think this would solve the problem, if I understand the code.
if `eventUser?.id` is null and `scopeUser.id` isn't null, the copied user will contain a null value.
merging should respect precedence, but if its null, we should use the one that has a value at least.

so I believe we could do something like:
`final id = eventUser?.id ?? scopeUser?.id`
for all properties, and then, create a new user, wdyt?

Reply:
Using `null` with a `copyWith` keeps the original value because it already does `other ?? this` itself.

Reply:
oh yeah, makes sense, my bad